benchmarks/tpcc/src/main/java/com/google/cloud/pgadapter/tpcc/JavaClientBenchmarkRunner.java

```diff
@@ -0,0 +1,257 @@
+      SpannerConfiguration spannerConfiguration,
+      Metrics metrics) {
+    super(statistics, tpccConfiguration, pgAdapterConfiguration, spannerConfiguration, metrics);
+  }
+
+  void setup() throws SQLException, IOException {
+    Spanner spanner = createSpanner();
+    spannerThreadLocal.set(spanner);
+    databaseClientThreadLocal.set(
+        spanner.getDatabaseClient(
+            DatabaseId.of(
+                spannerConfiguration.getProject(),
+                spannerConfiguration.getInstance(),
+                spannerConfiguration.getDatabase())));
+    transactionManagerThreadLocal.set(databaseClientThreadLocal.get().transactionManager());
+    Uninterruptibles.sleepUninterruptibly(Duration.ofMillis(2000L));
+  }
+
+  void teardown() throws SQLException {
+    transactionManagerThreadLocal.get().close();
+    spannerThreadLocal.get().close();
+  }
+
+  void executeStatement(String sql) throws SQLException {
+    if (sql.equals("begin transaction")) {
+      transactionThreadLocal.set(transactionManagerThreadLocal.get().begin());
+    } else if (sql.equals("commit")) {
+      try {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        transactionManagerThreadLocal.get().commit();
+        Duration executionDuration = stopwatch.elapsed();
+        metrics.recordLatency(executionDuration.toMillis());
+      } catch (AbortedException e) {
+        // Ignore the aborted exception. Roll back the transaction.
+        transactionManagerThreadLocal.get().rollback();
+      }
+    } else if (sql.startsWith("rollback")) {
+      transactionManagerThreadLocal.get().rollback();
+    } else if (sql.equals("set transaction read only")) {
+      // No-op because TransactionManager only supports read/write transactions.
+    }
+  }
+
+  void bindParams(Statement.Builder builder, Object[] params) throws SQLException {
+    for (int i = 0; i < params.length; i++) {
+      Object param = params[i];
+      if (param instanceof BigDecimal) {
+        builder.bind("p" + (i + 1)).to(Value.pgNumeric(((BigDecimal) param).toPlainString()));
+      } else if (param instanceof String) {
+        builder.bind("p" + (i + 1)).to((String) param);
+      } else if (param instanceof Integer) {
+        builder.bind("p" + (i + 1)).to((int) param);
+      } else if (param instanceof Long) {
+        builder.bind("p" + (i + 1)).to((long) param);
+      } else {
+        throw new SQLException(String.format("Unknown type for the parameter: %s", param));
+      }
+    }
+  }
+
+  Object[] paramQueryRow(String sql, Object[] params) throws SQLException {
+    ParametersInfo parametersInfo = convertPositionalParametersToNamedParameters(sql, '?', "$");
+    Statement.Builder builder = Statement.newBuilder(parametersInfo.sqlWithNamedParameters);
+    bindParams(builder, params);
+    Statement stmt = builder.build();
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    ResultSet resultSet = transactionThreadLocal.get().executeQuery(stmt);
```

Comment:
This does not actually execute the query. You will see a query execution time in the range of a few microseconds, or maybe even less, but that is because `executeQuery` does not actually execute the query. Instead, the query will only be executed when you call `ResultSet#next()`

Reply:
Yeah, I forgot this. Updated this.